monasca_events_api/tests/test_transforms.py

```diff
@@ -0,0 +1,347 @@
+    def test_on_post_fail_validate_authorization(
+            self,
+            _validate_authorization):
+        """Post Method fail due to validate authorization"""
+        _validate_authorization.side_effect = falcon.HTTPUnauthorized(
+            'Forbidden',
+            'Tenant does not have any roles')
+
+        transObj = TransformsSubClass()
+        transObj._transforms_repo = TransformsRepository()
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        try:
+            transObj.on_post(self._generate_req(), res)
+        except Exception as e:
+            self.assertRaises(falcon.HTTPUnauthorized)
+            self.assertEqual(e.status, '401 Unauthorized')
+
+    @mock.patch(
+        'monasca_events_api.v2.transforms.Transforms._create_transform_response')
+    @mock.patch(
+        'monasca_events_api.common.repositories.mysql.mysql_repository.mdb')
+    @mock.patch('monasca_events_api.openstack.common.uuidutils.generate_uuid')
+    @mock.patch(
+        'monasca_events_api.v2.transforms.Transforms._validate_transform')
+    @mock.patch(
+        'monasca_events_api.v2.common.helpers.validate_json_content_type')
+    @mock.patch(
+        'monasca_events_api.v2.transforms.Transforms._delete_transform')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_authorization')
+    @mock.patch('monasca_events_api.v2.common.helpers.get_tenant_id')
+    @mock.patch('monasca_events_api.v2.common.helpers.read_http_resource')
+    def test_on_post_pass(
+            self,
+            readhttp,
+            helper_tenant_id,
+            helpers_validate,
+            deleteTransform,
+            validjson,
+            validateTransform,
+            generateUUID,
+            mysqlRepo,
+            createRes):
+        """Post Method pass"""
+        helpers_validate.return_value = True
+        validjson.return_value = True
+        validateTransform.return_value = True
+        returnTransform = [{"region": "useast",
+                            "tenantId": "0ab1ac0a-2867-402d",
+                            "creation_time": "1434331190",
+                            "transform": "1"},
+                           {"region": "useast",
+                            "tenantId": "0ab1ac0a-2866-403d",
+                            "creation_time": "1234567890",
+                            "transform": "2"}]
+        createRes.return_value = returnTransform
+        readhttp.return_value = self._generate_req()
+        helper_tenant_id.return_value = '0ab1ac0a-2867-402d'
+        transform_id = "0ab1ac0a"
+        generateUUID.return_value = "067e6162-3b6f-4ae2-a171-2470b63dff00"
+
+        transObj = TransformsSubClass()
+        transObj._transforms_repo = TransformsRepository()
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        transObj.on_post(self._generate_req(), res)
+        self.assertEqual(falcon.HTTP_200, "200 OK")
```

Comment:
Same as the on_get this isn't necessarily valid transform data.  Try using real data and validating the res.body.